### PR TITLE
Search Functionality and Inject

### PR DESCRIPTION
**Pull request by Mctalian**
Adds Gulp Inject and Search Functionality

### CODE REVIEW

**Comment by Mctalian** [comment]
Requested Reviewers: @YashdalfTheGray, @bcbrennecke 

Please respond if I need to replace you.

We need to add David Parisi to this repo so we can tag him for code reviews.

**Comment by YashdalfTheGray** [comment]
I added Parisi to the repo, reviewing now.

File: src/ideas/searchFilter/searchFilter.spec.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      likes: 13
+    },
+    {
+      id: 4,
+      title: "Broken-Spacebar<Uses:Punctuation?Instead",
+      author: "Please*Fix&My%Keyboard$#@!()\'\"{}[]|\\/`~",
+      likes: 1
+    }
+  ];
+
```
Comment (YashdalfTheGray) [inline src/ideas/searchFilter/searchFilter.spec.js]: Fan of your test cases.

File: src/ideas/searchFilter/searchFilter.spec.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    {
+      id: 0,
+      title: "Idea One",
+      author: "Test User 1",
+      likes: 8
+    },
+    {
+      id: 1,
+      title: "Idea Dos",
+      author: "Karl",
```
Comment (YashdalfTheGray) [inline src/ideas/searchFilter/searchFilter.spec.js]: This made me think of [The Walking Dead](http://i.imgur.com/STdt5.jpg)

**Comment by YashdalfTheGray** [comment]
This looks good, job well done sir. Waiting on @bcbrennecke's approval now.

**Comment by Mctalian** [comment]
Adding @davethenipper in case @bcbrennecke cannot get to it.

**Comment by davethenipper** [comment]
Approved.  That's a hell of a delivery.

**Pull request merged**